x-pack/plugins/observability_solution/investigate_app/public/pages/details/components/add_from_library_button/index.tsx

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { OverlayRef } from '@kbn/core/public';
+import { v4 } from 'uuid';
+import { openAddPanelFlyout } from '@kbn/embeddable-plugin/public';
+import { i18n } from '@kbn/i18n';
+import type { PresentationContainer } from '@kbn/presentation-containers';
+import type { FinderAttributes } from '@kbn/saved-objects-finder-plugin/common';
+import React, { useMemo, useRef } from 'react';
+import { BehaviorSubject } from 'rxjs';
+import { useKibana } from '../../../../hooks/use_kibana';
+import { InvestigateTextButton } from '../investigate_text_button';
+
+// interface AddFromLibraryButtonProps {
+//   onWidgetAdd: (widget: InvestigateWidgetCreate) => Promise<void>;
+// }
+
+export function AddFromLibraryButton({ onWidgetAdd }) {
+  const children$ = useMemo(() => new BehaviorSubject({}), []);
+
+  const {
+    dependencies: {
+      start: { contentManagement },
+    },
+  } = useKibana();
+
+  const onWidgetAddRef = useRef(onWidgetAdd);
+
+  onWidgetAddRef.current = onWidgetAdd;
+
+  const panelRef = useRef<OverlayRef>();
+
+  const container = useMemo<
+    PresentationContainer & {
+      addNewEmbeddable: (
+        type: string,
+        explicitInput: { savedObjectId: string },
+        attributes: FinderAttributes
+      ) => Promise<{ id: undefined }>;
+    }
+  >(() => {
+    function addEmbeddable({
+      type,
+      title,
+      description,
+      attributes,
+      savedObjectId,
+    }: {
+      type: string;
+      title: string;
+      description?: string;
+      attributes: Record<string, any>;
+      savedObjectId: string;
+    }) {
+      const widget = {
+        id: v4(),
```

Comment:
@kdelemme Here's the id I had to add to make the embeddable render on the page. Without adding an id, I get following error coming from [this](https://github.com/elastic/kibana/blob/main/src/plugins/embeddable/public/add_panel_flyout/add_panel_flyout.tsx#L162) file

https://github.com/user-attachments/assets/1767fb76-2420-4a00-8bf1-ff03d9ed409c